### PR TITLE
Implement auto discovery

### LINKED ISSUE #115: Implement auto discovery
**Describe the feature you'd like**
The old Android app was able to autodiscover servers on the local network. I'd like to have this in the new app so I don't need to type the server address manually.

**Additional context**
I think apiclient developers created a function for that. 😉

### PR DESCRIPTION
**Pull request by CarlosOlivo**
Button only visible when at least one server is found

Closes #115 👀

### CODE REVIEW

**Comment by Maxr1998** [comment]
I love this, thank you so much! The only criticism I could spot in a quick review would probably be the standard AlertDialog with the newlines, but for now that's absolutely fine and in the future I may replace it with another UI anyways.
Gonna review it in detail later today, if I find the time.

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -2,6 +2,7 @@
     xmlns:tools="http://schemas.android.com/tools"
     package="org.jellyfin.mobile">
 
+    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
```
Comment (nielsvanvelzen) [inline app/src/main/AndroidManifest.xml:5]: Was this permission required for discovery to work? If so I'll add it to the apiclient for the next release.

**Review by nielsvanvelzen — COMMENTED** [review]
I don't think this is the correct approach in adding server discovery. In my opinion it should always start when the server selection screen is shown and display the results in the activity instead of using an dialog.

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (CarlosOlivo) [reply app/src/main/AndroidManifest.xml:5]: Yep, is required, also is already documented in the source code by ... you (?) https://github.com/jellyfin/jellyfin-apiclient-java/blame/1565f286a3b676cdf85093b0ee99c4b940b99660/android/src/main/kotlin/org/jellyfin/apiclient/discovery/AndroidBroadcastAddressesProvider.kt#L18

Comment (nielsvanvelzen) [reply app/src/main/AndroidManifest.xml:5]: It's funny how I knew it was required but forgot to actually add it to the manifest lol

**Comment by CarlosOlivo** [comment]
> I don't think this is the correct approach in adding server discovery. In my opinion it should always start when the server selection screen is shown and display the results in the activity instead of using an dialog.

Unfortunately for me the discovery of the cordova client never worked so, I don't know how it used to be the discovery UI, for me the UI is the login screen with a "Cancel" button that redirects to a no-op selectServer.html bundled in the web client (notice the height change in the title bar), but you are free to reimplement the UI to follow the original behavior / flow.

image
image

**Review by Maxr1998 — APPROVED** [review]

**Pull request merged**